### PR TITLE
docs(migration guide): update docs for 2.0.0 preview and migration

### PR DESCRIPTION
**Pull request by repjarms**
### Summary

This PR adds the preview information and transition guidelines for the upcoming `2.0.0` release.

### CODE REVIEW

**Review by mediumTaj — APPROVED** [review]
👍 looks good!

File: Migration-V2.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+```go
+service, err = assistantv1.NewAssistantV1(&assistantv1.AssistantV1Options{
+	Version:     core.StringPtr("2020-04-01"),
+	ServiceName: "assistant",
+ })
+```
+
+### Breaking Changes to model setter functions
+
+In the `1.x` branch of the SDK, setter methods were provided for almost all model structs for dealing with data from the service. Version `2.0.0` makes significant breaking changesin how setters are handled for these structs.
```
Comment (padamstx) [inline Migration-V2.md:28]:
```suggestion
In the `1.x` version of the SDK, setter methods were provided for almost all model structs for dealing with data from the service. Version `2.0.0` makes significant breaking changes in how setters are handled for these structs.
```

File: Migration-V2.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+Specifically, now the only structs that have setters for their properties are structs with the naming convention `{ServiceOperation}Options`. These are called `Options Models` and are typically the top level struct that is passed into a function. These structs have setter methods exposed that allow the individual properties within them to be set.
+
+For lower level models underneath these options models, they had setter methods for each individual property. As this bloated the code surface and provided little value compared to setting the property directly, we have decided in version `2.0.0` to remove these setters.
+
+If your code editor calls out that a setter method you are using has been removed, the appropriate way to update is to set the property directly.
+
+### Providing custom `Context` to functions
+
+Version `2.0.0` of the Go SDK now adds alternate versions to all service functions, named `{function}WithContext`. This function calls the expected service function, but accepts a `Context` as the first parameter. This allows you to provide a context that you would like the function to execute in.
```
Comment (padamstx) [inline Migration-V2.md:38]:
```suggestion
Version `2.0.0` of the Go SDK now includes an alternate version of each operation belonging to the service, named `<operation>WithContext`. This alternate function allows you to pass in a `context.Context` parameter while invoking the operation.  The `context.Context` parameter can be used to specify a timeout, or cancel an in-flight request.  Details about `context.Context` can be found [here](https://golang.org/pkg/context).
```

File: Migration-V2.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Internally, the Go SDK uses these functions for all operations, but the regular function provides the background context as a default.
```
Comment (padamstx) [inline Migration-V2.md:40]: This is a little bit confusing, but tbh I think I would just remove this sentence altogether.

File: Migration-V2.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+### Enabling `gzip` compression
+
+All service structs now expose a function `GetEnableGzipCompression()` and `SetEnableGzipCompression(enableGzip bool)`. These methods allow you to enable `gzip` compression within the service methods.
```
Comment (padamstx) [inline Migration-V2.md:44]: I would NOT mention these functions unless you know for a fact that watson services allow the client to send request bodies that are gzip-compressed.   This feature is typically enabled by setting the `x-gzip-compress-requests` extension in the API definition.  If that extension is set, then the service is generated such that gzip compression is enabled by default.  If it's not set, then we still generate the "enable gzip" function so the user could explicitly enable it, but it's not enabled by default.
So my recommendation is to not mention this at all.

File: Migration-V2.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+### Automatic Retries
+
+All service structs will now automatically retry HTTP methods in the event of an error.
```
Comment (padamstx) [inline Migration-V2.md:48]: This is actually not correct.   We now provide a mechanism to enable automatic retries, but they are not enabled by default.

File: Migration-V2.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+You can customize the retry logic to suit your purposes. The function `EnableRetries(maxRetries int, mayTretryInterval time.Duration)` allows you to configure the retry strategy that you would like to enable for the functions called by the service.
+
+Optionally, you can disable retries by using the `DisableRetries()` method exposed on the service struct.
```
Comment (padamstx) [inline Migration-V2.md:52]: Auto retries are disabled by default.

File: Migration-V2.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Service specific changes
+
+### assistantv1
+
+- In the `Message()` function, the MessageInput{}'s `Text` property is now a `*string`. This can be constructed via the `core.StringPtr("{message}")` convenience method.
```
Comment (padamstx) [inline Migration-V2.md:58]: FYI... the core.XXXPtr() methods are typically used for taking the address of a literal (e.g. `core.StringPtr("string literal")`) but if you have a variable in your app, then just take the address of it:
```
     var myMessage = "message text"
     newModel := &MessageInput{MessageText: &myMessage}
```

**Review by padamstx — CHANGES_REQUESTED** [review]
A few changes needed, plus a few other suggestions that should be considered.

**Comment by repjarms** [comment]
@padamstx I've updated the transition notes to address your comments and include your suggestions.

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**